src/include/platform/CHIPDeviceConfig.h

```diff
@@ -91,6 +91,52 @@
 #define CHIP_DEVICE_CONFIG_MAX_EVENT_QUEUE_SIZE 100
 #endif
 
+/**
+ * CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING
+ *
+ * Enable support for background event processing.
+ */
+#ifndef CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING
+#define CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING 1
+#endif
+
+
+/**
+ * CHIP_DEVICE_CONFIG_BG_TASK_NAME
+ *
+ * The name of the background task.
+ */
+#ifndef CHIP_DEVICE_CONFIG_BG_TASK_NAME
+#define CHIP_DEVICE_CONFIG_BG_TASK_NAME "BG"
+#endif
+
+/**
+ * CHIP_DEVICE_CONFIG_BG_TASK_STACK_SIZE
+ *
+ * The size (in bytes) of the background task stack.
+ */
+#ifndef CHIP_DEVICE_CONFIG_BG_TASK_STACK_SIZE
+#define CHIP_DEVICE_CONFIG_BG_TASK_STACK_SIZE (6*1024)
+#endif
+
+/**
+ * CHIP_DEVICE_CONFIG_BG_TASK_PRIORITY
+ *
+ * The priority of the background task.
+ */
+#ifndef CHIP_DEVICE_CONFIG_BG_TASK_PRIORITY
+#define CHIP_DEVICE_CONFIG_BG_TASK_PRIORITY 1
```

Comment:
This has to be properly set by callers. This should not be in top level CHIPDeviceConfig.h since this priority is FreeRTOS specific and would actually need to be set per product, and the feature only enabled if this is properly set.

Reply:
There are two other task priorities set in this file... if BG task priority doesn't go here, what about the others? They all have to play nicely together.

Reply:
I think we should figure out how we want to organize these configuration flags, and make that change in a subsequent PR.